src/libraries/System.Net.Http/src/System/Net/Http/HttpClient.cs

```diff
@@ -699,15 +734,23 @@ public void CancelPendingRequests()
 
         protected override void Dispose(bool disposing)
         {
-            if (disposing && !_disposed)
+            if (disposing)
             {
-                _disposed = true;
+                if (_isSharedInstance)
+                {
+                    return;
```

Comment:
As we're not disposing the client, we might see more issues with H/3 similar to this: https://github.com/dotnet/runtime/issues/71927
The problem is that without Dispose, msquic might not send the last frames before exiting the app, causing problems on the other side, e.g. missing ACKs causing hangs.